integrations/watsonx/src/haystack_integrations/components/generators/watsonx/chat/chat_generator.py

```diff
@@ -0,0 +1,391 @@
+
+        merged_kwargs.pop("stream", None)
+
+        return {"messages": watsonx_messages, "params": merged_kwargs}
+
+    def _handle_streaming(
+        self,
+        api_args: dict[str, Any],
+        callback: StreamingCallbackT,
+    ) -> dict[str, list[ChatMessage]]:
+        """Handle synchronous streaming response."""
+        chunks: list[StreamingChunk] = []
+        stream = self.client.chat_stream(messages=api_args["messages"], params=api_args["params"])
+
+        for chunk in stream:
+            if not isinstance(chunk, dict) or not chunk.get("choices"):
+                continue
+
+            content = chunk["choices"][0].get("delta", {}).get("content", "")
+            if content:
+                chunk_meta = {
+                    "model": self.model,
+                    "index": chunk["choices"][0].get("index", 0),
+                    "finish_reason": chunk["choices"][0].get("finish_reason"),
+                    "received_at": datetime.now(timezone.utc).isoformat(),
+                }
+                streaming_chunk = StreamingChunk(content=content, meta=chunk_meta)
+                chunks.append(streaming_chunk)
+                callback(streaming_chunk)
+
+        return {"replies": [self._convert_streaming_chunks_to_chat_message(chunks)]}
+
+    def _handle_standard(self, api_args: dict[str, Any]) -> dict[str, list[ChatMessage]]:
+        """Handle synchronous standard response."""
+        response = self.client.chat(messages=api_args["messages"], params=api_args["params"])
+        return self._process_response(response)
+
+    async def _handle_async_streaming(
+        self,
+        api_args: dict[str, Any],
+        callback: StreamingCallbackT,
+    ) -> dict[str, list[ChatMessage]]:
+        """Handle asynchronous streaming response."""
+        chunks: list[StreamingChunk] = []
+        stream_generator = await self.client.achat_stream(messages=api_args["messages"], params=api_args["params"])
+
+        async for chunk in stream_generator:
+            if not isinstance(chunk, dict) or not chunk.get("choices"):
+                continue
+
+            content = chunk["choices"][0].get("delta", {}).get("content", "")
+            if content:
+                chunk_meta = {
+                    "model": self.model,
+                    "index": chunk["choices"][0].get("index", 0),
+                    "finish_reason": chunk["choices"][0].get("finish_reason"),
+                    "received_at": datetime.now(timezone.utc).isoformat(),
+                }
+                streaming_chunk = StreamingChunk(content=content, meta=chunk_meta)
+                chunks.append(streaming_chunk)
+                try:
+                    result = callback(streaming_chunk)
+                    if asyncio.iscoroutine(result):
+                        await result
+                except Exception:
+                    # Fallback to sync call if async fails
+                    callback(streaming_chunk)
```

Comment:
This should just be 
```suggestion
                await callback(streaming_chunk)
```
since we don't allow a sync version of the callback to be passed.